### PR TITLE
Migrate 'components/get_ios_app' module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by AGMETEOR**
**Summary**

MM-18959-Migrate 'components/get_ios_app' module and associated tests to TypeScript

**Ticket Link**

[https://mattermost.atlassian.net/browse/MM-18959](https://mattermost.atlassian.net/browse/MM-18959)

Fixes https://github.com/mattermost/mattermost-server/issues/12468

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -5167,6 +5167,12 @@
           "resolved": "https://registry.npmjs.org/regexp-tree/-/regexp-tree-0.1.13.tgz",
           "integrity": "sha512-hwdV/GQY5F8ReLZWO+W1SRoN5YfpOKY6852+tBFcma72DKBIcHjPRIlIvQN35bCOljuAfP2G2iB0FC/w236mUw==",
           "dev": true
+        },
+        "semver": {
```
Comment (devinbinnie) [inline package-lock.json]: Can you revert your `package-lock.json` changes? They aren't needed for the TypeScript migration.

Comment (AGMETEOR) [reply package-lock.json]: @devinbinnie I made the revert. Not sure why these changes occurred, any idea why?

Comment (devinbinnie) [reply package-lock.json]: Whenever `npm install` is run, `package-lock.json` regenerates and moves/add/removes things. 
I'm not sure why to be honest. My personal strategy is to just not commit the file unless I'm updating/adding a specific package.

Comment (AGMETEOR) [reply package-lock.json]: That's a good one. Thinking about it now, I think I need to ditch my extravagant use of `git add .` :)

**Pull request merged**